### ISSUE TITLE
Linux：Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by Allen188** [vdemichev/DiaNN#23]
I have 6,000 DIA to run, so I had to run multiple computers on Linux. So far, I have tested two samples and reported an error: “Segmentation fault (core dumped)”, and the result file cannot be generated. The test parameters are as follows. 
diann-linux --f A20181202sunyt_shaoyk_CRCA_DIA_b21_8.mzML --f A20181202sunyt_shaoyk_CRCA_DIA_b21_9.mzML --lib G20190417chenlb_38IDA_spn.xls --threads 20 --verbose 1 --out "report.tsv" --out-gene "report.genes.tsv" --qvalue 0.01 --fasta swissprot_human_20180209_target_IRT_contaminant.fasta --individual-mass-acc --individual-windows --rt-profiling --pg-level 1 
In addition, I have two questions, 1. Can you provide all the parameters of the Linux version ?2. Can the results obtained from multiple computers be directly combined.

**Comment by vdemichev** [comment]
Hi Allen188,

For me to be able to figure out why it does not work on those runs, could you please share the full DIA-NN output (starting from the command you use to run DIA-NN and ending with the error) generated at log level 3 ("--verbose 3" command)? Please also use the latest DIA-NN version of GitHub.

1. Parameters for the Linux version are exactly the same as for the Windows version. If you are not sure what commands to use on Linux, I would suggest configuring DIA-NN on Windows first using the GUI and then using the same commands on Linux as those printed by GUI.

2. Yes, it's possible. One can either 
(1) Run analyses on several PCs, then put all the .quant files together in one folder (e.g. /home/temp/) and generate a report using these .quant files (--use-quant option in combination with --temp /home/temp/). Would still need to specify the raw files using --f [file name], but these raw files will not be accessed by DIA-NN if there are .quant files with the respective names available.

(2) Simply merge final reports in R. However, in the latter case selection of fragment ions for quantification should be disabled or performed using --restrict-fr (manual, page 4) instead of automatic cross-run selection. Also, in this case one should perform normalisation manually, instead of relying on DIA-NN's cross-run normalisation: use Precursor.Quantity instead of Precursor.Normalised.

Btw, we ourselves have run a 6000+ sample experiment, so DIA-NN is perfectly fine with such analyses, although of course it might take a bit longer on a single PC. 

Hope this helps!

Best wishes,
Vadim

**Comment by vdemichev** [comment]
Please reopen this issue once you have more information on the error.

**Comment by jspaezp** [comment]
Hello there!

I am working on another large run and I got a core-dump due to a dda file sneaking into my file list, it might be a place for improvement to check if the file is actually DIA and throw a more informative error message

```
DIA-NN 1.8.1 (Data-Independent Acquisition by Neural Networks)
Compiled on Apr 15 2022 08:45:18
Current date and time: Fri Aug  4 22:26:32 2023
Logical CPU cores: 64
Thread number set to 12
Scan windows will be inferred separately for different runs
Only peaks with correlation sum exceeding 2 will be considered
Peaks with correlation sum below 1 from maximum will not be considered
A single score will be used until RT alignment to save memory; this can potentially lead to slower search
Mass accuracy will be fixed to 1.3e-05 (MS2) and 5e-06 (MS1)

1 files will be processed
[0:00] Loading spectral library lib.predicted.speclib
[0:13] Library annotated with sequence database(s): uniprot_human_sp_isoforms_2022-03-15_crap.fasta
[0:13] Gene names missing for some isoforms
[0:13] Library contains 20338 proteins, and 20137 genes
[0:17] Spectral library loaded: 42318 protein isoforms, 70426 protein groups and 9508661 precursors in 3596349 elution groups.
[0:17] Initialising library

[0:26] File #1/1
[0:26] Loading run SOMERANDOMRUN_DDA.d
For most diaPASEF datasets it is better to manually fix both the MS1 and MS2 mass accuracies to values in the range 10-15 ppm.
.command.sh: line 13:   204 Segmentation fault      (core dumped) diann --lib lib.predicted.speclib --f SOMERANDOMRUN_DDA.d --threads 12 --verbose 3 --individual-windows --temp ./ --min-corr 2.0 --corr-diff 1.0 --mass-acc 13 --mass-acc-ms1 5 --time-corr-only 2>&1
       206 Done                    | tee SOMERANDOMRUN_DDA_diann.log

```

(so rather than a bug, this could be a feature request for more informative error messages when a DDA file is passed)

Best!
Sebastian
